tools/Google.Cloud.Tools.ReleaseManager/CreateClientsCommand.cs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Cloud.Tools.Common;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Google.Cloud.Tools.ReleaseManager;
+
+public class CreateClientsCommand : CommandBase
+{
+    private const string PublishTargetFramework = "netstandard2.1";
+
+    public CreateClientsCommand()
+        : base("create-clients", "Checks that we can create clients for a pure-GAPIC package", "id")
+    {
+    }
+
+    protected override void ExecuteImpl(string[] args)
+    {
+        string id = args[0];
+        var catalog = ApiCatalog.Load();
+        if (!catalog.TryGetApi(id, out var api))
+        {
+            throw new UserErrorException($"No such API: {id}");
+        }
+        if (api.Generator != GeneratorType.Micro)
+        {
+            Console.WriteLine($"{id} is not a GAPIC-generated API; skipping client creation test.");
+        }
+
+        var assembly = PublishAndLoadAssembly(id);
+        if (assembly is null)
+        {
+            return;
+        }
+        CreateClients(assembly, id);
+    }
+
+    // Note: a lot of code from here on is copied/modified from SmokeTest and SuggestSmokeTestsCommand.
+    // We could consider refactoring later, if we find this is a problem.
+    private Assembly PublishAndLoadAssembly(string id)
+    {
+        var sourceRoot = DirectoryLayout.ForApi(id).SourceDirectory;
+        Processes.RunDotnet(sourceRoot, "publish", "-nologo", "-clp:NoSummary", "-v", "quiet", "-c", "Release", id, "-f", PublishTargetFramework);
+
+        var assemblyFile = Path.Combine(sourceRoot, id, "bin", "Release", PublishTargetFramework, "publish", $"{id}.dll");
+        return Assembly.LoadFrom(assemblyFile);
+    }
+
+    private void CreateClients(Assembly assembly, string id)
+    {
+        var clientTypes = assembly.GetTypes()
+            .Where(t => t.IsClass)
+            .Where(t => t.Name.EndsWith("Client"))
+            .Where(t => assembly.GetType(t.FullName + "Builder") is Type builderType
+                        && builderType.BaseType.Name == "ClientBuilderBase`1")
+            .ToList();
+
+        if (clientTypes.Count == 0)
+        {
+            throw new UserErrorException($"API {id} has no client types.");
+        }
```

Comment:
Can we just do nothing here? Thinking of the log proto packages, etc., so we don't have to filter them out in bash possibly?

Reply:
They'll be filtered out earlier, by not using the GAPIC generator. (I believe, anyway. I'm running build-everything run now.)

Reply:
Ha... that was the *plan*, anyway. But:

```text
Google.Apps.Script.Type is not a GAPIC-generated API; skipping client creation test.
Configuration error: API Google.Apps.Script.Type has no client types.
```

Reply:
(The joys of a missing return statement.)